make-go.bash

```diff
@@ -50,10 +66,20 @@ if [ -d "$GOROOT_FINAL" ]; then
 fi
 mkdir -p "$GOROOT_FINAL"
 
+# Fetch latest Go
+if [ ! -f go-latest.tar.gz ]; then
+  wget -O go-latest.tar.gz $GO_SOURCE_URL$GO_SOURCE
```

Comment:
The issue I see when you use a hardcoded archive name is the following:
* The script is run, downloads go 1.6.3 and saves it as go-latest.tar.gz
* The script gets update to use go 1.6.4 --> the sha1 hash changes as well
* go-latest.tar.gz that was previously downloaded and contains go 1.6.3 will now make the build script stop because of a checksum mismatch.